yarn-project/prover-node/src/prover-node.ts

```diff
@@ -1,73 +1,148 @@
 
   constructor(
-    private prover: EpochProverManager,
-    private publisher: L1Publisher,
-    private l2BlockSource: L2BlockSource,
-    private l1ToL2MessageSource: L1ToL2MessageSource,
-    private contractDataSource: ContractDataSource,
-    private worldState: WorldStateSynchronizer,
-    private coordination: ProverCoordination,
-    private simulator: SimulationProvider,
-    private telemetryClient: TelemetryClient,
+    private readonly prover: EpochProverManager,
+    private readonly publisher: L1Publisher,
+    private readonly l2BlockSource: L2BlockSource,
+    private readonly l1ToL2MessageSource: L1ToL2MessageSource,
+    private readonly contractDataSource: ContractDataSource,
+    private readonly worldState: WorldStateSynchronizer,
+    private readonly coordination: ProverCoordination,
+    private readonly simulator: SimulationProvider,
+    private readonly quoteProvider: QuoteProvider,
+    private readonly quoteSigner: QuoteSigner,
+    private readonly claimsMonitor: ClaimsMonitor,
+    private readonly epochsMonitor: EpochMonitor,
+    private readonly telemetryClient: TelemetryClient,
     options: Partial<ProverNodeOptions> = {},
   ) {
     this.options = {
       pollingIntervalMs: 1_000,
-      disableAutomaticProving: false,
       maxPendingJobs: 100,
-      epochSize: 2,
       ...compact(options),
     };
 
     this.metrics = new ProverNodeMetrics(telemetryClient, 'ProverNode');
   }
 
+  async ensureBond() {
+    // Ensure the prover has enough bond to submit proofs
+    // Can we just run this at the beginning and forget about it?
+    // Or do we need to check periodically? Or only when we get slashed? How do we know we got slashed?
+  }
+
+  async handleClaim(proofClaim: EpochProofClaim): Promise<void> {
+    if (proofClaim.epochToProve === this.latestEpochWeAreProving) {
+      this.log.verbose(`Already proving claim for epoch ${proofClaim.epochToProve}`);
+      return;
+    }
+
+    try {
+      await this.startProof(proofClaim.epochToProve);
+      this.latestEpochWeAreProving = proofClaim.epochToProve;
+    } catch (err) {
+      this.log.error(`Error handling claim for epoch ${proofClaim.epochToProve}`, err);
+    }
+  }
+
+  /**
+   * Handles the epoch number to prove when the prover node starts by checking if there
+   * is an existing claim for it. If not, it creates and sends a quote for it.
+   * @param epochNumber - The epoch immediately before the current one when the prover node starts.
+   */
+  async handleInitialEpochSync(epochNumber: bigint): Promise<void> {
+    try {
+      const claim = await this.publisher.getProofClaim();
+      if (!claim || claim.epochToProve < epochNumber) {
+        await this.handleEpochCompleted(epochNumber);
+      } else if (claim && claim.bondProvider.equals(this.publisher.getSenderAddress())) {
+        const lastEpochProven = await this.l2BlockSource.getProvenL2EpochNumber();
+        if (!lastEpochProven || lastEpochProven < claim.epochToProve) {
```

Comment:
should be `if (lastEpochProven === undefined || ...)` I think.

Also, we will probably want a configuration to check that we aren't waking up and trying to prove too late. Probably needs to be a config param that is our estimated minimum time needed to prove, and we don't even try if we don't have that much time left in the epoch. Certainly not needed at the moment. I filed https://github.com/AztecProtocol/aztec-packages/issues/8957

Reply:
Makes sense! I'll address the zero vs undefined check in another PR so as not to lose the green CI on this one :-P